fuzzy-native/src/MatcherBase.cpp

```diff
@@ -35,12 +35,51 @@ inline string str_to_lower(const std::string &s) {
   return lower;
 }
 
+float distance_to_root(const std::string *root, const std::string *value) {
+  size_t num_common_dirs = 0;
+  size_t i = 0;
+
+  // Count number of common directories
+  for (; i < root->length() && i < value->length(); i++) {
+    if (root->at(i) != value->at(i)) {
```

Comment:
`->at` comes at a performance penalty (but the benefit is range checking).
Here it's pretty obvious that we won't exceed the bounds so array access is preferable.